### DISCUSSION TITLE
support css post-processing loaders when use experiments.css

### DISCUSSION THREAD

**Discussion by 9aoy** [webpack/webpack#17320] [Q&A] [answered]
## Feature request

**What is the expected behavior?**

Can use some css post-processing loaders (like [css-modules-typescript-loader](https://github.com/seek-oss/css-modules-typescript-loader))，when I enable experiments.css.

**What is motivation or use case for adding/changing the behavior?**
when I use css-loader, I can use `css-modules-typescript-loader` to create TypeScript declarations for CSS Modules.

This loader needs to be executed after css-loader because it needs to access the result of css-modules processing.

But, when I use experiments.css, I can't declare this loader after experiments.css processing.

```
module.exports = {
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          'css-modules-typescript-loader',
          {
            loader: 'css-loader',
            options: {
              modules: true
            }
          }
        ]
      }
    ]
  }
};
```

**How should this be implemented in your opinion?**

**Are you willing to work on this yourself?**
yes

**Comment by alexander-akait** [comment]
Set `type: "javascript/auto"`, you can't use basic `css` feature  with `css-loader`, because it generates javascript, it is incompatibility with how it works, feel free to update our docs and feel free to feedback

**Reply by 9aoy** [reply]
Is there any way to add post-processing loaders when using experiments.css?

**Reply by 9aoy** [reply]
Based on my understanding, currently the processing of CSS when using experiments.css is not done in the loader stage, so it is not possible to add post-processing loaders. 

Perhaps an ability can be added to experiments.css to declare its use as a builtin loader. Then I can configure it like this:

```
module.exports = {
  module: {
    experiments: {
       css: {
         mode: 'loader',
       },
    },
    rules: [
      {
        test: /\.css$/,
        type: 'css/module',
        use: [
          'css-modules-typescript-loader',
         'builtin:css'
        ]
      }
    ]
  }
};
```

**Reply by alexander-akait** [reply, marked as answer]
What do you mean postprocessing? After render css you have assets and you can do anything with it, `css-modules-typescript-loader` doesn't have compatibility with new CSS pipeline, it should be a plugin, because every CSS file is a module with dependecies, so now you don't need to parse/execute code, just get CSS modules and filter dependecies and you can create types

**Reply by 9aoy** [reply]
Okay, now I understand what you mean. I can implement this feature through a plugin. Thanks.
